app/scripts/background.js

```diff
@@ -308,22 +308,16 @@ function maybeDetectPhishing(theController) {
       // blocking is better than tab redirection, as blocking will prevent
       // the browser from loading the page at all
       if (isManifestV2) {
-        if (details.type === 'sub_frame') {
-          // redirect the entire tab to the
-          // phishing warning page instead.
-          redirectTab(details.tabId, redirectHref);
-          // don't let the sub_frame load at all
-          return { cancel: true };
-        }
-        // redirect the whole tab
-        return { redirectUrl: redirectHref };
+        // redirect the whole tab (even if it's a sub_frame request)
+        redirectTab(details.tabId, redirectHref);
+        return { cancel: true };
```

Comment:
In the first version of this code, we only delt with `main_frame` and `sub_frame`, so the implicit `else` branch was always handling `main_frame`. The idiomatic way of cancelling and redirecting the tab is to do `return { redirectUrl }`. After removing the request type (`main_frame`, `sub_frame`) filter the `return { redirectUrl }` path because the default for non-`sub_frame` requests.

I'm concerned there might be a UX degradation by switching to the asynchronous `redirectTab` function for `main_frame`, vs the previous way which utilizes MV2's (Firefox's) _blocking_ feature. I know it is still `cancel`led synchronously, but I'm just wondering if combining `cancel` with the async redirect is not the best way to go about it.

Is there a problem with using `return { redirectUrl }` for the `main_frame`? Do you think my concerns are unwarranted (I don't feel very strongly about changing `main_frame` handling back, it just feels like the correct solution in this case)?

Reply:
Thats warranted feeback i would say, however in my opinion I think we should go with the proposed solution as it now follows the approach as MV3 allowing for better consistency and less chances of something weird happening  but we still get the bonus of canceling the requests that is only available in MV2.